src/wccp2.cc

```diff
@@ -2024,10 +2033,16 @@ wccp2AssignBuckets(void *)
                                     tmp_rtr,
                                     &wccp_packet,
                                     offset);
+                    // XXX: setup fd_table[] properly so we can use FD_WRITE_METHOD()
+                    fd_table[theWccp2Connection->fd].bytesWritten(offset);
                 } else {
-                    if (send(theWccp2Connection, &wccp_packet, offset, 0) < static_cast<int>(offset)) {
+                    const auto x = send(theWccp2Connection, &wccp_packet, offset, 0);
+                    if (x < static_cast<int>(offset)) {
```

Comment:
We can simplify and avoid (paranoid) 64-to-32 bit value truncation concerns:

```suggestion
                    if (x < 0) {
```